hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonFileInputFormat.java

```diff
@@ -88,6 +99,50 @@ public CarbonTable getOrCreateCarbonTable(Configuration configuration) throws IO
     }
   }
 
+  /**
+   * This method will list all the carbondata files in the table path and treat one carbondata
+   * file as one split.
+   */
+  public List<InputSplit> getAllFileSplits(JobContext job) throws IOException {
```

Comment:
Don't add more public methods. Please take a conf property from outside and decide whether to store in datamap or just list splits plainly like this.